### ISSUE TITLE
`ind_1` tests needs to be changed /added

### ISSUE THREAD

**Issue by simonedelsa** [CORE-forge/coresoi#10]
Hi there!

As far as ind_1 is concerned, for each company we have the following 2x2 table

|      | 0   | 1   |    |
|------|-----|-----|----|
| pre  | n11 | n12 | m1 |
| post | n21 | n22 | m2 |

where column variable is our *flag_division* (CPV33 no/yes).

Here are some comments:
- `compute_fisher` function should be applied considering `alternative="greater"`

```r
compute_fisher <- function(a, b, c, d) {
  data <- matrix(c(a, b, c, d), ncol = 2)
  fish <- fisher.test(data, alternative = "greater)
  c(
    p_value = round(fish$p.value, 3),
    fisher_estimate = round(fish$estimate, 3)
  )
}
```

-  when it calls Fisher test, order of data should be changed (even if it shouldn't affect the test result), as follows:    
    
```r
      fisher_test = compute_fisher(n_11, n_21, n_12, n_22)[[1]],
      fisher_estimate = compute_fisher(n_11, n_21, n_12, n_22)[[2]],
```

- unconditional Barnard test --> here is the code:

```r
compute_barnard <- function(a, b, c, d) {
  data <- matrix(c(a, b, c, d), ncol = 2)
  barn <- Exact::exact.test(data, alternative = "greater", 
                            to.plot = FALSE, method = "Boschloo")
  c(
    p_value = round(barn$p.value, 3),
    barn_estimate = round(barn$estimate, 3)
  )
}
```

and contextually in the `ind_1` function body the following rows should be added (order of data intentionally changed wrt Fisher test) after computing the Fisher test:

```r
      barnard_test = compute_barnard(n_22, n_12, n_21, n_11)[[1]],
      barnard_estimate = compute_barnard(n_22, n_12, n_21, n_11)[[2]],
      barnard_test = round(barnard_test, 3)
```

**Comment by NiccoloSalvini** [comment]
Barnard ([this implementation in R](https://andrisignorell.github.io/DescTools/reference/BarnardTest.html) @simonedelsa this is different from the package tou chose i.e. Exact which required older and more dependencies) test has several merits against Fisher test such that:

- non-parametric
- more powerful under some specifications

but it is more time-consuming to compute (References can be found in the [Wikipedia article](https://en.wikipedia.org/wiki/Barnard%27s_test) on the subject).

``` r
library(coresoi)
library(microbenchmark)
data("mock_data_core")


results <- microbenchmark(
  with_barnard = ind_1(
    data = mock_data_core %>%  head(10000), 
    publication_date = data_pubblicazione, 
    cpv = cod_cpv,
    emergency_name = "coronavirus",
    stat_unit = provincia,
    barnard = TRUE),
  without_barnard  = ind_1(
    data = mock_data_core %>% head(10000), 
    publication_date = data_pubblicazione, 
    cpv = cod_cpv,
    emergency_name = "coronavirus",
    stat_unit = provincia,
    barnard = FALSE),
  times = 10)
results
```

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:////Users/niccolo/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip.htm">
<link rel=File-List
href="file:////Users/niccolo/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip_filelist.xml">

</head>

<body link="#0563C1" vlink="#954F72">



expr | min | lq | mean | median | uq | max | neval | cld
-- | -- | -- | -- | -- | -- | -- | -- | --
with_barnard | 2473,342 | 2509,767 | 2.588 | 2.535 | 264 | 2.851 | 10 | b
without_barnard | 70,185 | 7154479,000 | 126 | 119 | 175 | 217 | 10 | a



</body>

</html>


<sup>Created on 2023-01-30 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

I implemented a logical argument to select if the user would like to wait for Barnard or just have Fisher. DO we really need to stick to that?. I also found [this article](https://www.researchgate.net/publication/226516333_Simplifying_the_calculation_of_the_P-value_for_Barnard's_test_and_its_derivatives) which tries to solve the computationally complexity limits for Barnard, no code found btw.

**Comment by simonedelsa** [comment]
Ok @NiccoloSalvini, we were aware of that. Just for curiosity, if you have time, as we won't compute the indicator for 'stat_unit = provincia' but, rather, for the winner companies...could you 'microbenchmark' with this setting?

**Comment by NiccoloSalvini** [comment]
implemented different test choice option for `ind_1` and `ind_2` in 619812bdfc67ea3e087ad0fd1953d36397cdf285.
The others will follow!

**Comment by NiccoloSalvini** [comment]
>Ok @NiccoloSalvini, we were aware of that. Just for curiosity, if you have time, as we won't compute the indicator for 'stat_unit = provincia' but, rather, for the winner companies...could you 'microbenchmark' with this setting?


Based on 619812bdfc67ea3e087ad0fd1953d36397cdf285 I revised the code according to the newer version of `ind_1`( merged #15 ) for which we have to specify which test we want to perform `test_type = "barnard"`

```r
library(coresoi)
library(microbenchmark)
data("mock_data_core")


results <- microbenchmark(
  with_barnard = ind_1(
    data = mock_data_core %>%  head(10000), 
    publication_date = data_pubblicazione, 
    cpv = cod_cpv,
    emergency_name = "coronavirus",
    stat_unit = cf_amministrazione_appaltante,
    test_type =  "barnard"),
  without_barnard  = ind_1(
    data = mock_data_core %>% head(10000), 
    publication_date = data_pubblicazione, 
    cpv = cod_cpv,
    emergency_name = "coronavirus",
    stat_unit = cf_amministrazione_appaltante,
    test_type =  "fisher"),
  times = 10)
results
```

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:////Users/niccolo/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip.htm">
<link rel=File-List
href="file:////Users/niccolo/Library/Group%20Containers/UBF8T346G9.Office/TemporaryItems/msohtmlclip/clip_filelist.xml">
</head>

<body link="#0563C1" vlink="#954F72">



expr | min | lq | mean | median | uq | max | neval | cld
-- | -- | -- | -- | -- | -- | -- | -- | --
with_barnard | 5.033,00 | 5.044,15 | 5.075,04 | 5.077,82 | 5.106,53 | 5.122,02 | 10 | b
with_barnard | 182,35 | 182,79 | 184,38 | 183,62 | 186,40 | 187,03 | 10 | a



</body>

</html>
